drivers/Ticker.h

```diff
@@ -54,26 +58,20 @@ namespace mbed {
  * }
  *
  * int main() {
- *     timer.attach(&attime, 5);
+ *     timer.attach(&attime, 5us);
  *     while(1) {
  *         if(flip == 0) {
  *             led1 = !led1;
  *         } else {
  *             led2 = !led2;
  *         }
- *         wait(0.2);
+ *         ThisThread::sleep_for(200ms);
  *     }
  * }
  * @endcode
  */
-class Ticker : public TimerEvent, private NonCopyable<Ticker> {
-
+class TickerBase : public TimerEvent {
```

Comment:
`TimerEvent` is noncopyable, so dropped the specification from all its derived classes. They inherit the noncopyableness.

I guess I should be retaining it for the classes that need it themselves - ie the ones that have destructors. Make explicit that that class must be noncopyable, because it's managing something, not relying on the base class being noncopyable. So `TickerBase` should have it, `Ticker` needn't.

I guess you could restate the non-copyable for documentation purposes on `Ticker`, but then by the same argument you'd want to restate all the inherited public methods too...